w3af/core/data/url/handlers/gzip_handler.py

```diff
@@ -57,7 +57,7 @@ def http_response(self, request, response):
                     data = gzip.GzipFile(fileobj=StringIO(body)).read()
                 elif "deflate" in enc_hdr:
                     body = response.read()
-                    data = zlib.decompress(body)
+                    data = zlib.decompress(body, -zlib.MAX_WBITS)
```

Comment:
I'm afraid of this line introducing a regression in the gzip handler. What if this fixes the case you're facing but breaks other cases? The zlib window size, as explained in the [python docs](https://docs.python.org/2/library/zlib.html#zlib.decompress) is a little bit difficult to understand and I'm not sure if this change will allow w3af to decompress other bodies.

Maybe the best bet is to a) try to decompress using `data = zlib.decompress(body)` and if that fails with the exception `Error -3 while decompressing data: incorrect header check` (be specific when catching this exception) we try b) `data = zlib.decompress(body, -zlib.MAX_WBITS)`

Do you think this would work?